docs/archive-data.md

```diff
@@ -0,0 +1,314 @@
+    // This method should thrift errors - see filestore as an example.
+    Get(context.Context, URI, *GetHistoryRequest) (*GetHistoryResponse, error)
+
+    // ValidateURI is used to define what a valid URI for an implementation is.
+    ValidateURI(URI) error
+}
+```
+
+```go
+type VisibilityArchiver interface {
+    // Archive is used to archive one workflow visibility record.
+    // Check the Archive() method of the HistoryArchiver interface in Step 2 for parameters' meaning and requirements.
+    // The only difference is that the ArchiveOption parameter won't include an option for recording process.
+    // Please make sure your implementation is lossless. If any in-memory batching mechanism is used, then those batched records will be lost during server restarts.
+    // This method will be invoked when workflow closes. Note that because of conflict resolution, it is possible for a workflow to through the closing process multiple times, which means that this method can be invoked more than once after a workflow closes.
+    Archive(context.Context, URI, *ArchiveVisibilityRequest, ...ArchiveOption) error
+
+    // Query is used to retrieve archived visibility records.
+    // Check the Get() method of the HistoryArchiver interface in Step 2 for parameters' meaning and requirements.
+    // The request includes a string field called query, which describes what kind of visibility records should be returned. For example, it can be some SQL-like syntax query string.
+    // Your implementation is responsible for parsing and validating the query, and also returning all visibility records that match the query.
+    // Currently the maximum context timeout passed into the method is 3 minutes, so it's ok if this method takes a long time to run.
+    Query(context.Context, URI, *QueryVisibilityRequest) (*QueryVisibilityResponse, error)
+
+    // ValidateURI is used to define what a valid URI for an implementation is.
+    ValidateURI(URI) error
+}
+```
+
+### Update provider
+
+Update provider to enable access to your implementation by modifying the [`/provider/provider.go`](https://github.com/temporalio/temporal/blob/master/common/archiver/provider/provider.go) file so that the ArchiverProvider knows how to create an instance of your archiver. 
+
+### Add configs
+
+Add configs for you archiver to static yaml config files, such as `config/development.yaml`. 
+
+Then modify the [`HistoryArchiverProvider`](https://github.com/temporalio/temporal/blob/master/common/service/config/config.go#L403) and [`VisibilityArchiverProvider`](https://github.com/temporalio/temporal/blob/master/common/service/config/config.go#L421) struct in the `./common/service/config.go` accordingly.
+
+### Custom archiver FAQ
+
+**If my Archive method can automatically be retried by the caller, how can I record and access progress between retries?**
```

Comment:
at least from the guide itself I don't understand what this problem actually means

Reply:
Anyone going through the custom archiver steps will understand once they try to implement the methods